NitroxClient/MonoBehaviours/RemotePlayerColorApplicator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using NitroxClient.GameLogic;
+using NitroxClient.GameLogic.PlayerModel;
+using NitroxClient.GameLogic.PlayerModel.Abstract;
+using NitroxClient.GameLogic.PlayerModel.ColorSwap;
+using NitroxModel.Core;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours
+{
+    public class RemotePlayerColorApplicator : MonoBehaviour
+    {
+        private IEnumerable<IColorSwapManager> colorSwapManagers;
+        private RemotePlayer remotePlayer;
+        private ColorSwapAsyncOperation swapOperation;
+
+        public void Awake()
+        {
+            colorSwapManagers = NitroxServiceLocator.LocateService<IEnumerable<IColorSwapManager>>();
+        }
+
+        public void Update()
+        {
+            if (remotePlayer == null)
+            {
+                return;
+            }
+
+            if (swapOperation != null)
+            {
+                return;
+            }
+
+            swapOperation = new ColorSwapAsyncOperation(remotePlayer, colorSwapManagers);
+            StartCoroutine(ApplyPlayerColor());
+        }
+
+        public void AttachRemotePlayer(RemotePlayer remotePlayer)
+        {
+            if (this.remotePlayer != null)
+            {
+                throw new InvalidOperationException("An applicator instance may only be used once.");
+            }
+
+            this.remotePlayer = remotePlayer;
+        }
+
+        private IEnumerator ApplyPlayerColor()
+        {
+            swapOperation.BeginColorSwap();
+
+            while (!swapOperation.IsColorSwapComplete())
+            {
+                yield return new WaitForSeconds(0.1f);
+            }
+
+            swapOperation.ApplySwappedColors();
+
+            Destroy(this);
```

Comment:
Your point is a fair one. I generally have a hard time taking static references like that. However; automated testing is pretty much out of the question given this level of coupling with Unity to begin with, so this is probably a good example of when it would be best for me to keep my snotty opinions to myself.

I'll clean that up. This just felt more like the "Unity way" of doing things, but I honestly don't know how efficient this is compared to just using a component that is already running.